main/messages.h

```diff
@@ -0,0 +1,7 @@
+// Simple struct for a dynamically sized send message
+
+typedef struct send_message
+{
+    int32_t msg_length;
+    uint8_t *buffer;
```

Comment:
we could optionally add the arbitration_id the request is going to/response is coming from if you wanted